packages/cpp/src/browser/cpp-commands.ts

```diff
@@ -11,41 +11,81 @@ import { CommandContribution, CommandRegistry, Command } from '@theia/core/lib/c
+import { DirNode, FileDialogFactory, FileStatNode } from '@theia/filesystem/lib/browser';
 import { EditorManager } from "@theia/editor/lib/browser";
+import { FileSystem } from '@theia/filesystem/lib/common';
+import { WorkspaceService } from '@theia/workspace/lib/browser/workspace-service';
 
-/**
- * Switch between source/header file
- */
-export const SWITCH_SOURCE_HEADER: Command = {
-    id: 'switch_source_header',
-    label: 'Switch between source/header file'
-};
+export namespace Commands {
+
+    export const OPEN: Command = {
+        id: 'cpp:open',
+        label: 'Specify folder location...'
+    };
+
+    /**
+     * Switch between source/header file
+     */
+    export const SWITCH_SOURCE_HEADER: Command = {
+        id: 'switch_source_header',
+        label: 'Switch between source/header file'
+    };
 
-export const FILE_OPEN_PATH = (path: string): Command => <Command>{
-    id: `file:openPath`
-};
+    /**
+     * Switch between source/header file
+     */
+    export const CHANGE_CONFIGURATION: Command = {
+        id: 'change_configuration',
+        label: 'Change Clangd Configuration'
+    };
+
+}
+
+export interface DidChangeConfigurationParams {
+    /**
+    * The actual changed settings
+    */
+    settings: ClangdConfigurationParams;
+}
+
+export interface ClangdConfigurationParams {
+    /**
+    * The compilation database path
+    */
+    compilationDatabasePath?: string;
+}
 
 @injectable()
 export class CppCommandContribution implements CommandContribution {
 
     constructor(
+        @inject(FileSystem) protected readonly fileSystem: FileSystem,
+        @inject(FileDialogFactory) protected readonly fileDialogFactory: FileDialogFactory,
         @inject(CppClientContribution) protected readonly clientContribution: CppClientContribution,
         @inject(OpenerService) protected readonly openerService: OpenerService,
         @inject(EditorManager) private editorService: EditorManager,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService,
         protected readonly selectionService: SelectionService
 
     ) { }
 
     registerCommands(commands: CommandRegistry): void {
-        commands.registerCommand(SWITCH_SOURCE_HEADER, {
+        commands.registerCommand(Commands.SWITCH_SOURCE_HEADER, {
             isEnabled: () => (this.editorService && !!this.editorService.activeEditor &&
                 (this.editorService.activeEditor.editor.document.uri.endsWith(".cpp") || this.editorService.activeEditor.editor.document.uri.endsWith(".h"))),
             execute: () => {
                 this.switchSourceHeader();
             }
         });
+        commands.registerCommand(Commands.CHANGE_CONFIGURATION, {
+            isEnabled: () => (this.editorService && !!this.editorService.activeEditor &&
+                (this.editorService.activeEditor.editor.document.uri.endsWith(".cpp") || this.editorService.activeEditor.editor.document.uri.endsWith(".h"))),
```

Comment:
Checking just .h and .cpp is no good, we need many more extension , let's fix this in another patch.

Reply:
This should be generally configurable...